### ISSUE TITLE
Blazor Wasm - NotifyAuthenticationStateChanged within DelegatingHandler doesn't work

### ISSUE THREAD

**Issue by ducman** [dotnet/aspnetcore#28610]
### Describe the bug
I'm unsure if this is intended but I'm using a custom DelegatingHandler to put a bearer token on the request header (similar to the built-in AuthenticationMessageHandler). In this handler, I'm calling NotifyAuthenticationStageChanged whether the user is now authorized or no longer authorized yet my UI remains unchanged.

### To Reproduce
From unauthorized to authorized, navigate to a page (unsecured) that on initialized makes a call to an API that requires a custom delegating handler which always notifies authentication state changed that the user is now authorized. You'll notice that any component with AuthorizeView/Authorized isn't rendered. If you take out the API call and just notify authentication state changed that the user is now authorized, this works.

From authorized to unauthorized, navigate to a screen that on initialized makes a call to an API that requires a custom delegating handler which always notifies authentication stage changed that the user is no longer authorized. You'll notice that any component with AuthorizeView/Authorized isn't hidden (or rerendered to be removed). If I navigate to an authorized screen (has authorize attribute), I'm not redirected to the login screen. If you take out the API call and just notify authentication state changed that the user is not authorized, this works.

It's something to do with this call being inside of the delegating handler. 

### Further technical details
.NET SDK (reflecting any global.json):
 Version:   5.0.100
 Commit:    5044b93829

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.100\

Host (useful for support):
  Version: 5.0.0
  Commit:  cf258a14b7

.NET SDKs installed:
  1.1.11 [C:\Program Files\dotnet\sdk]
  2.1.202 [C:\Program Files\dotnet\sdk]
  2.1.503 [C:\Program Files\dotnet\sdk]
  2.1.505 [C:\Program Files\dotnet\sdk]
  2.2.102 [C:\Program Files\dotnet\sdk]
  3.0.100 [C:\Program Files\dotnet\sdk]
  3.1.102 [C:\Program Files\dotnet\sdk]
  3.1.201 [C:\Program Files\dotnet\sdk]
  5.0.100 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.0.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.9 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us.
It's not clear why you're trying to build your own implementation here. If you want to pursue that path, we recommend asking this question on [StackOverflow](https://stackoverflow.com). Otherwise, if you believe there is  a framework functionality which doesn't work as expected, please share with us a small repro project and indicate what exactly misbehaves.

**Comment by ducman** [comment]
We have an SSO authentication model that doesn't exactly fit the ones provided/in the templates, hence, you can see why. But we're not too far off from the built-in implementation. I'm just trying to figure out a global way to determine whether the user is no longer authenticated (i.e. token expired) **before** making an API call. I know making the API call will tell you it's a 401 but that's a lot of boilerplate code (try/catch, if/else) around each API call which is why I threw it in a `DelegatingHandler`. 

Unfortunately, I've spent too much time trying to figure what's going wrong here to get an example repo for you to see. When I get some time, I'll throw one together.

**Comment by ducman** [comment]
So I think I found the underlining issue. It seems the instance of the custom `AuthenticationStateProvider` that is injected into the custom `DelegatingHandler` is not the same as the one injected onto a page (if you were to inject one onto a page, etc.). That's why `NotifyAuthenticationStateChanged` on this other instance doesn't really trigger a re-rendering of any `AuthorizeView` component.

I have created a repo [here ](https://github.com/ducman/BlazorCustomAuth5) to show that any `Scoped` service that is injected into a `DelegatingHandler` behave more like a transient unlike how it is documented [here ](https://docs.microsoft.com/en-us/aspnet/core/blazor/fundamentals/dependency-injection?view=aspnetcore-5.0&pivots=webassembly) that it is a `Singleton`: 

Adding the service as a `Singleton` does fix the issue. So for Blazor wasm, should we avoid `Scoped` and just stick with `Singleton` and `Transient`?
